gapic/templates/%namespace/%name_%version/%sub/services/%service/transports/rest.py.j2

```diff
@@ -0,0 +1,200 @@
+            scopes: Sequence[str] = None,
+            ssl_channel_credentials: grpc.ChannelCredentials = None,
+            quota_project_id: Optional[str] = None,
+            client_info: gapic_v1.client_info.ClientInfo = DEFAULT_CLIENT_INFO,
+            ) -> None:
+        """Instantiate the transport.
+
+        Args:
+            host ({% if service.host %}Optional[str]{% else %}str{% endif %}):
+                {{- ' ' }}The hostname to connect to.
+            credentials (Optional[google.auth.credentials.Credentials]): The
+                authorization credentials to attach to requests. These
+                credentials identify the application to the service; if none
+                are specified, the client will attempt to ascertain the
+                credentials from the environment.
+            
+            credentials_file (Optional[str]): A file with credentials that can
+                be loaded with :func:`google.auth.load_credentials_from_file`.
+                This argument is ignored if ``channel`` is provided.
+            scopes (Optional(Sequence[str])): A list of scopes. This argument is
+                ignored if ``channel`` is provided.
+            ssl_channel_credentials (grpc.ChannelCredentials): SSL credentials
+                for grpc channel. It is ignored if ``channel`` is provided.
+            quota_project_id (Optional[str]): An optional project to use for billing
+                and quota.
+            client_info (google.api_core.gapic_v1.client_info.ClientInfo):	
+                The client info used to send a user-agent string along with	
+                API requests. If ``None``, then default info will be used.	
+                Generally, you only need to set this if you're developing	
+                your own client library.
+        """
+        super().__init__(host=host, credentials=credentials)
+        self._session = AuthorizedSession(self._credentials)
+    {%- if service.has_lro %}
+
+    @property
+    def operations_client(self) -> operations_v1.OperationsClient:
+        """Create the client designed to process long-running operations.
+
+        This property caches on the instance; repeated calls return the same
+        client.
+        """
+        # Sanity check: Only create a new client if we do not already have one.
+        if 'operations_client' not in self.__dict__:
+            from google.api_core import grpc_helpers
+            self.__dict__['operations_client'] = operations_v1.OperationsClient(
+                grpc_helpers.create_channel(
+                    self._host,
+                    credentials=self._credentials,
+                    scopes=self.AUTH_SCOPES,
+                )
+            )
+
+        # Return the client from cache.
+        return self.__dict__['operations_client']
+    {%- endif %}
+
+
+    {%- for method in service.methods.values() %}
+    {# TODO(yonmg): consider using enums #}
+        {%- set ns = namespace(http=None) %}
```

Comment:
Is namespace a jinja class?

Reply:
Yes. It allows variable assignment to carry through a loop body since apparently just using the `{%- set %}` tag and a variable loses any changes once it leaves for loop scope.

Reply:
Neat! TIL